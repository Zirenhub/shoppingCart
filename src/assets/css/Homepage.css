*,
*::before,
*::after {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.homepage-container {
  padding-top: 120px;
  min-height: 100vh;
	background: linear-gradient(-45deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
	background-size: 400% 400%;
	animation: gradient 15s ease infinite;
}

@keyframes gradient {
	0% {
		background-position: 0% 50%;
	}
	50% {
		background-position: 100% 50%;
	}
	100% {
		background-position: 0% 50%;
	}
}

.homepage-title {
  font-size: 2rem;
  padding-left: 20px;
}

h1 > span {
  color: #ffffff;
}

.homepage-content p:nth-child(1) {
  font-size: 1.4rem;
  letter-spacing: 5px;
  padding-left: 50px;
}

.homepage-content p:nth-child(1) > span {
  color: #ffffff;
}

.homepage-content p:nth-child(2) {
  font-size: 1.8rem;
  letter-spacing: 6px;
  padding-left: 370px;
  color: #ffffff;
}

.homepage-content p:nth-child(3) {
  font-size: 1.5rem;
  letter-spacing: 9px;
  padding-left: 70px;
}

.homepage-content p:nth-child(3) > span {
  color: #ffffff;
}

.shop-button {
  display: flex;
  justify-content: center;
  align-items: center;
}

.shop-button > a > button {
  padding: 5px 30px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  background-color: transparent;
  color: #ffffff;
  cursor: pointer;
  transition: ease-in-out 200ms;
}

.shop-button > a > button:hover {
  border: 1px solid rgba(0, 0, 0, 0.5);
  transform: scale(1.1);
}


