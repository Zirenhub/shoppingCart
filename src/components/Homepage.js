const Homepage = () => {
  return (
    <div className="homepage-container">
      <h1>This is homepage!</h1>
    </div>
  );
};

export default Homepage;
