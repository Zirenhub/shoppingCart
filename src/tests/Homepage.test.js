import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
// import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Router } from 'react-router-dom';
import Homepage from '../components/Homepage';
import { createMemoryHistory } from '@remix-run/router';

describe('Homepage renders', () => {
  const history = createMemoryHistory();

  test('Shop Button renders', () => {
    render(
      <Router location={history.location} navigator={history}>
        <Homepage />
      </Router>
    );

    const button = screen.getByRole('button', {
      name: /Shop Now/i,
    });

    expect(button).toBeInTheDocument();
  });

  test('Matches DOM Snapshot', () => {
    const domTree = renderer
      .create(
        <Router location={history.location} navigator={history}>
          <Homepage />
        </Router>
      )
      .toJSON();

    expect(domTree).toMatchSnapshot();
  });
});
