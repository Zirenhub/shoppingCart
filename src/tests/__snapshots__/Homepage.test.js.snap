// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders Matches DOM Snapshot 1`] = `
<div
  className="homepage-container"
>
  <div
    className="welcome-message"
  >
    <div
      className="homepage-title"
    >
      <h1>
        Welcome to the online store that 
        <span>
          delivers
        </span>
      </h1>
    </div>
    <div
      className="homepage-content"
    >
      <div>
        <p>
          Only the most 
          <span>
            high quality
          </span>
        </p>
        <p>
          fresh
        </p>
        <p>
          fast-fashion 
          <span>
            clothing
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="shop-button"
  >
    <a
      href="/shop"
      onClick={[Function]}
    >
      <button>
        Shop Now
      </button>
    </a>
  </div>
</div>
`;
